internal/controllers/topology/cluster/util.go

```diff
@@ -37,6 +37,11 @@ func infrastructureMachineTemplateNamePrefix(clusterName, machineDeploymentTopol
 	return fmt.Sprintf("%s-%s-infra-", clusterName, machineDeploymentTopologyName)
 }
 
+// infrastructureMachinePoolNamePrefix calculates the name prefix for a InfrastructureMachinePool.
+func infrastructureMachinePoolNamePrefix(clusterName, machinePoolTopologyName string) string {
+	return fmt.Sprintf("%s-%s-infra-", clusterName, machinePoolTopologyName)
```

Comment:
```suggestion
	return fmt.Sprintf("%s-%s-", clusterName, machinePoolTopologyName)
```

Sorry was just discussing this with Fabrizio in #9135, can you already change it?

Reply:
Got it, should I be making this change for other name prefixes in `util.go`, or just my current diff?

Reply:
Just for yours. I change all others in a separate PR